smoketest/scripts/cli/test_qos.py

```diff
@@ -922,6 +922,236 @@ def test_20_round_robin_policy_default(self):
+            rec = rec.strip()
+            if 'root' in rec:
+                self.assertEqual(rec, 'qdisc hfsc 1: root refcnt 2 default 2')
+            else:
+                self.assertRegex(
+                    rec,
+                    r'qdisc sfq \S+: parent 1:2 limit 127p quantum 1514b depth 127 flows 128 divisor 1024 perturb 10sec',
+                )
+        # use raw because tc json is incorrect here
+        tmp = cmd(f'tc -details class show dev {interface}')
+        for rec in tmp.split('\n'):
+            rec = rec.strip().lower()
+            if 'root' in rec:
+                self.assertEqual(rec, 'class hfsc 1: root')
+            elif 'hfsc 1:1' in rec:
+                # m2 \S+bit is auto bandwidth
+                self.assertRegex(
+                    rec,
+                    r'class hfsc 1:1 parent 1: sc m1 0bit d 0us m2 \S+bit ul m1 0bit d 0us m2 \S+bit',
+                )
+            else:
+                self.assertRegex(
+                    rec,
+                    rf'class hfsc 1:2 parent 1:1 leaf \S+: ls m1 0bit d 0us m2 {ls["m2"]} ul m1 {ul["m1"]} d {ul["d"]}ms m2 {ul["m2"]}',
+                )
+
+        for key, val in rt.items():
+            self.cli_set(
+                base_path
+                + [
+                    'policy',
+                    'shaper-hfsc',
+                    policy_name,
+                    'default',
+                    'realtime',
+                    key,
+                    val,
+                ]
+            )
+        self.cli_commit()
+
+        tmp = cmd(f'tc -details class show dev {interface}')
+        for rec in tmp.split('\n'):
+            rec = rec.strip().lower()
+            if 'hfsc 1:2' in rec:
+                self.assertTrue(
+                    f'rt m1 {rt["m1"]} d {rt["d"]}ms m2 {rt["m2"]} ls m1 0bit d 0us m2 {ls["m2"]} ul m1 {ul["m1"]} d {ul["d"]}ms m2 {ul["m2"]}'
+                    in rec
+                )
+
+        # add some class
+        self.cli_set(
+            base_path
+            + [
+                'policy',
+                'shaper-hfsc',
+                policy_name,
+                'class',
+                '10',
+                'linkshare',
+                'm2',
+                '300kbit',
+            ]
+        )
+        self.cli_set(
+            base_path
+            + [
+                'policy',
+                'shaper-hfsc',
+                policy_name,
+                'class',
+                '10',
+                'match',
+                'tst',
+                'ip',
+                'dscp',
+                'internet',
+            ]
+        )
+
+        self.cli_set(
+            base_path
+            + [
+                'policy',
+                'shaper-hfsc',
+                policy_name,
+                'class',
+                '30',
+                'realtime',
+                'm2',
+                '250kbit',
+            ]
+        )
+        self.cli_set(
+            base_path
```

Comment:
Please do not bload the config because ruff says so.

Rather use a sane abstraction like foo_path =  base_path + ['policy', 'shaper-hfsc']

Reply:
fixed